lib/src/nips/nip_004/event.dart

```diff
@@ -45,8 +45,24 @@ class EncryptedDirectMessage extends Event {
     return ciphertext;
   }
 
-  String getPlaintext(String receiverPrivkey, String senderPubkey) {
-    return Nip4.decipher(receiverPrivkey, senderPubkey, content);
+  String getPlaintext(String receiverPrivkey, [String senderPubkey=""]) {
```

Comment:
Decrypting an event should always use  the sender pub key provided in the event (not in the tags). But I left `senderPubkey` as an optional parameter.

Other thing is, if the decryption fails, there is no way to indicate that to the caller. I think `getPlaintext()` should throw an exception when decryption fails. @ethicnology do you have an opinion on that?

Reply:
I did this refactoring based on the signature of your function decrypt but you are right, this should be as simple as possible so we should  probably remove this param